scripts/git-hooks/pre-commit

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set -e
+
+formatFileList="$(git --no-pager diff --name-status --no-color --cached | awk '$1 != "D" && $2 ~ /\.kts|\.kt/ { print $NF}')"
+
+echo "spotlesscheck start..."
+for filePath in $formatFileList
+do
+  ./gradlew spotlessApply -PspotlessIdeHook="$(pwd)/$filePath"
```

Comment:
It can be slow 👀 
How about using ratchet?

```
if(isHook) {
  ratchetFrom 'origin/main'
}
```

https://github.com/diffplug/spotless/tree/main/plugin-gradle#how-can-i-enforce-formatting-gradually-aka-ratchet